src/app/js/lib/getPercentage.js

```diff
@@ -0,0 +1,11 @@
+import get from 'lodash.get';
+
+export function getPercentValue(data, decimals = 0) {
+    const weight = get(data, 'weight', 0);
```

Comment:
It could be better to pass directly the weight. In fact,  the data object is a little bit too specific.